### ISSUE TITLE
False positives for google_checks when using switch expression in lambda

### ISSUE THREAD

**Issue by Edarke** [checkstyle/checkstyle#16762]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Example.java
/var/tmp $ java -jar ~/Downloads/google-java-format-1.26.0-all-deps.jar --replace Example.java; cat Example.java 
import java.util.Optional;

public class Example {

  /** Main method. */
  public static String main(Optional<Integer> opt) {
    return opt.map(
            n ->
                switch (n) {
                  case 1 -> "Test";
                  default -> "Default";
                })
        .orElse("");
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $  java -jar ~/Downloads/google-java-format-1.26.0-all-deps.jar --replace Example.java; cat Example.java;  java $RUN_LOCALE -jar ~/Downloads/checkstyle-10.23.0-all.jar -c /google_checks.xml Example.java 
Starting audit...
[WARN] /Users/evan.darke/Documents/IntellijProjects/Mongot2/Example.java:3:1: Missing a Javadoc comment. [MissingJavadocType]
[WARN] /Users/evan.darke/Documents/IntellijProjects/Mongot2/Example.java:9:17: 'switch' has incorrect indentation level 16, expected level should be one of the following: 6, 8, 12. [Indentation]
[WARN] /Users/evan.darke/Documents/IntellijProjects/Mongot2/Example.java:10:19: 'case' child has incorrect indentation level 18, expected level should be one of the following: 8, 10, 14. [Indentation]
[WARN] /Users/evan.darke/Documents/IntellijProjects/Mongot2/Example.java:11:19: 'case' child has incorrect indentation level 18, expected level should be one of the following: 8, 10, 14. [Indentation]
[WARN] /Users/evan.darke/Documents/IntellijProjects/Mongot2/Example.java:12:17: 'switch rcurly' has incorrect indentation level 16, expected level should be one of the following: 6, 8, 12. [Indentation]
Audit done.
```

---
This expands on https://github.com/checkstyle/checkstyle/issues/15592, which partially fixed behavior for switch expressions.

I know google_java_format is maintained separately than Checkstyle. I know there are multiple interpretations of google java style. I know there's no guarantee that these tools are compatible. However, I also think the output prescribed by Checkstyle (shown below) in this particular case is objectively an incorrect interpretation of google style. The switch expression is a continuation of the previous line and should be indented.

```java
import java.util.Optional;

public class Example {

  /** Main method. */
  public static String main(Optional<Integer> opt) {
    return opt.map(
            n ->
            switch (n) {
              case 1 -> "Test";
              default -> "Default";
            })
        .orElse("");
  }
}
```
---

Currently, I'm using the following workaround, but it's not ideal as it broadly disables the check.

```
 <module name="SuppressionXpathSingleFilter">
      <property name="files" value=".*.java"/>
      <property name="checks" value="Indent"/>
      <property name="query"
        value="//(ASSIGN | EXPR)/LITERAL_SWITCH
        | //(ASSIGN | EXPR)/LITERAL_SWITCH/SWITCH_RULE/LITERAL_CASE
        | //(ASSIGN | EXPR)/LITERAL_SWITCH/SWITCH_RULE/LITERAL_DEFAULT
        | //(ASSIGN | EXPR)/LITERAL_SWITCH/SWITCH_RULE/SLIST//*
       | //(ASSIGN | EXPR)/LITERAL_SWITCH/RCURLY "/>
</module>
```

**Comment by romani** [comment]
Yes, we try to keep compatibility with Google formatter.
Google  formatter is part of our test suite.

**Comment by Abdelrhmansersawy** [comment]
@romani , Does this code snippet violate the Google Java Style Guide?
```java
return opt.map(
        n ->
          switch (n) {         // violation: incorrect indentation level 10, expected 12
            case 1 -> "Test";   // violation: incorrect indentation level 12, expected 14
            default -> "Default"; // violation: incorrect indentation level 12, expected 14
          })                // violation: incorrect indentation level 8, expected 12
        .orElse("");
```

I tried fixing the issue, and the solution works correctly with this formatting:
```java
return opt.map(
        n ->
            switch (n) {
              case 1 -> "Test";
              default -> "Default";
            })
        .orElse("");
```

I'm wondering - is the first snippet compliant with the Google Java Style Guide or not?

**Comment by Zopsss** [comment]
Yes, I think the indentation should be increase by +4 based on this rule: https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent

did you used [google-java-formatter](https://github.com/google/google-java-format) to fix the code? if not can you verify it using this tool just to confirm we're heading in the correct direction.

**Comment by romani** [comment]
Should be fixed by https://github.com/checkstyle/checkstyle/pull/16721

**Comment by romani** [comment]
Fixed by https://github.com/checkstyle/checkstyle/issues/15098

**Comment by Edarke** [comment]
Hi @romani, thanks for looking into this. I just tried this use case on 10.24 and I still see similar violations. I can confirm the linked case `testMethod3` does work. From what I can tell, the key difference depends on whether`.map` gets line wrapped or not.

10.24 output on original case:

```bash
java -jar google-java-format-1.27.0-all-deps.jar --replace Example.java && cat -n Example.java  && java $RUN_LOCALE -jar ~/Downloads/checkstyle-10.24.0-all.jar -c /google_checks.xml Example.java
     1	import java.util.Optional;
     2	
     3	/** Example. */
     4	public class Example {
     5	
     6	  /** Main method. */
     7	  public static String main(Optional<Integer> opt) {
     8	    return opt.map(
     9	            n ->
    10	                switch (n) {
    11	                  case 1 -> "Test";
    12	                  default -> "Default";
    13	                })
    14	        .orElse("");
    15	  }
    16	
    17	  String testMethod3(List<Integer> operations) {
    18	    return operations.stream()
    19	        .map(
    20	            op ->
    21	                switch (op) {
    22	                  case 1 -> "test";
    23	                  default -> "TEST";
    24	                })
    25	        .findFirst()
    26	        .orElse("defaultValue");
    27	  }
    28	}
Starting audit...
[WARN] /Users/evan.darke/Downloads/Example.java:10:17: 'switch' has incorrect indentation level 16, expected level should be one of the following: 10, 12. [Indentation]
[WARN] /Users/evan.darke/Downloads/Example.java:11:19: 'case' child has incorrect indentation level 18, expected level should be one of the following: 12, 14. [Indentation]
[WARN] /Users/evan.darke/Downloads/Example.java:12:19: 'case' child has incorrect indentation level 18, expected level should be one of the following: 12, 14. [Indentation]
[WARN] /Users/evan.darke/Downloads/Example.java:13:17: 'switch rcurly' has incorrect indentation level 16, expected level should be one of the following: 10, 12. [Indentation]
Audit done.
```

10.24 on modified case with line wrap:

```bash
 java -jar google-java-format-1.27.0-all-deps.jar --replace Example1.java && cat -n Example1.java  && java $RUN_LOCALE -jar ~/Downloads/checkstyle-10.24.0-all.jar -c /google_checks.xml Example1.java
     1	import java.util.Optional;
     2	
     3	/** Example. */
     4	public class Example1 {
     5	
     6	  /** Main method. */
     7	  public static String main(Optional<Integer> integerOptional) {
     8	    return integerOptional
     9	        .map(
    10	            n ->
    11	                switch (n) {
    12	                  case 1 -> "Test";
    13	                  default -> "Default";
    14	                })
    15	        .orElse("");
    16	  }
    17	
    18	  String testMethod3(List<Integer> operations) {
    19	    return operations.stream()
    20	        .map(
    21	            op ->
    22	                switch (op) {
    23	                  case 1 -> "test";
    24	                  default -> "TEST";
    25	                })
    26	        .findFirst()
    27	        .orElse("defaultValue");
    28	  }
    29	}
Starting audit...
Audit done.
```
